src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -258,8 +256,7 @@ object ArrayBuffer extends StrictOptimizedSeqFactory[ArrayBuffer] {
     val k = coll.knownSize
     if (k >= 0) {
       val array = new Array[AnyRef](k max DefaultInitialSize)
-      val it = coll.iterator
-      for (i <- 0 until k) array(i) = it.next().asInstanceOf[AnyRef]
+      IterableOnce.copyElemsToArray(coll, array.asInstanceOf[Array[Any]])
```

Comment:
For the specific case of an `ArrayBuffer`, and since we have arrays, do we need a polymorphic iterator? Perhaps it would be a bit quicker to do `newArray(i) := oldArray(i)`

Reply:
I don't understand your question. I changed it to specifically not use an iterator unless necessary